apollo-api/src/main/java/com/apollographql/apollo/api/Input.java

```diff
@@ -59,7 +59,7 @@ private Input(V value, boolean defined) {
 
     Input<?> input = (Input<?>) o;
 
-    return defined == input.defined && value != null && value.equals(input.value);
+    return defined == input.defined && value == input.value;
```

Comment:
Thank you for your feedback.
As long as I know, using `==` works in case that both values are null too. I wrote some tests trying to cover all possible scenarios using a `String` as well as a custom `TestObject` class as values for the `Input` and all of them seems to work with my approach.
Please let me know if I am missing something.

Reply:
Indeed, `==` works for null value as well, I read the opposite change. But still, doing `value == input.value` only checks for reference equality. I believe we want value equality.

Reply:
You are absolutely right.

I updated the code and added your example as a test.

Thanks!!